### PR TITLE
Add atomic file operation for symlink changes

### LINKED ISSUE #62768: [FEATURE REQUEST] add atomic file operation for symlink changes
**Is your feature request related to a problem? Please describe.**
There is a narrow gap between link removal and link creation, which can be problematic for real time workloads

**Describe the solution you'd like**
Allow symlink functions to use atomic file operations (create temporary link and move into place after creation)

**Describe alternatives you've considered**
N/A

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Introduces the `atomic` parameter to the `file.symlink` execution and states modules

### What issues does this PR fix or reference?
Fixes: #62768

### Previous Behavior
Narrow gap between link removal and creation during link changes and force mode

### New Behavior
Move operation is atomic when enabled

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by nicholasmhughes** [comment]
re-run pr-freebsd-131-amd64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-freebsd-131-amd64-py3-pytest

**Comment by nicholasmhughes** [comment]
re-run pr-freebsd-131-amd64-py3-pytest

**Review by Ch3LL — COMMENTED** [review]
There is actually salt.utils.atomicfile funciton. Would that help here?

**Comment by nicholasmhughes** [comment]
That function only handles opening files for atomic write. I briefly considered adding parts of this logic to that util, but the swip-swapping I'm doing here just doesn't seem to fit well.

File: salt/states/file.py [hunk 1]
```diff
@@ -1747,12 +1753,9 @@ def symlink(
 
     if __salt__["file.is_link"](name):
         # The link exists, verify that it matches the target
-        if os.path.normpath(__salt__["file.readlink"](name)) != os.path.normpath(
+        if os.path.normpath(__salt__["file.readlink"](name)) == os.path.normpath(
             target
         ):
-            # The target is wrong, delete the link
-            os.remove(name)
```
Comment (Ch3LL) [inline salt/states/file.py:1754]: why are you removing the logic to remove the link here?

File: salt/states/file.py [hunk 2]
```diff
@@ -1832,14 +1835,7 @@ def symlink(
                         name, backupname, exc
                     ),
                 )
-        elif force:
```
Comment (Ch3LL) [inline salt/states/file.py:1835]: same question here. Why are you removing the logic to force the removal of link if force is set.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/states/file.py (hunk 2 above)
Comment (nicholasmhughes) [reply salt/states/file.py:1835]: that logic is handled in the execution module. note that I'm passing `force` to it, whereas it wasn't there before

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**